src/main/webapp/js/index.js

```diff
@@ -32,22 +30,23 @@ async function loadLogoutButton(account) {
       `You are signed in as ${account.email}`;
 }
 
-/** Fetches all receipts from the server and adds them to the DOM. */
-async function getAllReceipts() {
+/**
+ * Fetches all receipts from the server and adds them to the DOM.
+ * This will only be called on page load.
+ */
+function getAllReceipts() {
```

Comment:
nit: With pagination, it's no longer "all" receipts right? I would rename this and edit the comment.